### DISCUSSION TITLE
Add letsblock.it as valid location for subscription links

### DISCUSSION THREAD

**Discussion by xvello** [uBlockOrigin/uBlock-issues#1987] [Ideas] [answered]
Hello, https://letsblock.it maintainer here!

The LBI project allows users to generate a custom filter list, by picking and customizing community-maintained filter templates. Its ambition is not to replace the current filter list ecosystem, but augment it when a one-size-fits-all approach does not work.

I have seen a very good reception for the project, but want to work on reducing the setup friction, with a tighter integration with the uBlock Origin extension. I forked uBO and created a commit to [enable subscription links](https://github.com/xvello/uBlock/commit/c8d1834018fba0de0c4aac2088be12e6ff379515) from `letsblock.it` and `staging.letsblock.it`, but cannot open a PR on the project. Could you please guide me on how to get this commit upstream?

Also would also love to discuss how we could make manually refreshing a list easier. If I setup my [search results filter](https://letsblock.it/filters/search-results) to exclude one more website, I have to trigger a list refresh for the change to be effective. Ideally, I'd love for it to be as easy as the subscription link. How would you feel about adding such a feature?

**Comment by gorhill** [comment]
For now I would prefer you use the `abp:subscribe?location=[...]` trick on `https://letsblock.it/` and tell visitors they can right-click to subscribe to the list. On GitHub itself, you can use the `https://subscribe.adblockplus.org/?location=[...]` trick. This latter trick also work on non-trusted sites in the current dev build to subscribe to the list through contextual menu.

**Reply by xvello** [reply]
Thanks for your reply! I updated the install instruction to use an `abp:` link: 
![image](https://user-images.githubusercontent.com/6241083/154713900-737566e5-a171-4af2-ab13-fc69616d2680.png)

**Reply by xvello** [reply]
@gorhill unfortunately, the context menu trick does not work on Firefox Mobile (on Android), while the click handler on whitelisted domains works fine.
Do you know whether that can be fixed in a future extension release (different API to register a context menu item), or whether that's a limitation of extensions on mobile?

**Comment by gorhill** [comment, marked as answer]
> whether that's a limitation of extensions on mobile?

Limitation on Firefox for Android, it [does not support context menus](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/menus#browser_compatibility). Given the [reception on HN](https://news.ycombinator.com/item?id=30057442), I will import your changes to allow easy subscription through one click.

**Reply by xvello** [reply]
Thanks! It works great with `1.41.7b0`. 

I was double-guessing whether the `"https://*.letsblock.it/*"` pattern would correctly match `https://letsblock.it/help/use-list`, but it works fine.

**Comment by ghost** [comment]
Wiki updated with this new site.

https://github.com/uBlockOrigin/uBlock-issues/wiki/Dashboard:-Filter-lists/_compare/fc69cf4cb2a9382dc8c938b8ada3e7d241078663

**Comment by gorhill** [comment]
@xvello For your interest, I have added a new procedural cosmetic filter operator which purpose is to make it easier to create reader-mode filters, see <https://github.com/gorhill/uBlock/commit/152120bd9ec7a2ccea907e015fb195484ef7bc8e>. I think this kind of operator is a good match for your online tool.
